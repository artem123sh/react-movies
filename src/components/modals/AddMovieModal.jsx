import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Modal from '../basic/Modal';
import Button from '../basic/Button';
import MovieForm from '../forms/MovieForm';

const StyledActionContainer = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-top: 3rem;
`;

const StyledButton = styled(Button)`
  margin-left: 1rem;
`;

const AddMovieModal = ({
  onClose, handleMovieAdd,
}) => (
  <Modal size="small" title="add movie" onClose={onClose}>
    <MovieForm onSubmit={handleMovieAdd}>
      <StyledActionContainer>
        <StyledButton variant="outlined" size="large" type="reset">Reset</StyledButton>
        <StyledButton variant="contained" size="large" type="submit">Submit</StyledButton>
      </StyledActionContainer>
    </MovieForm>
  </Modal>
);

AddMovieModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  handleMovieAdd: PropTypes.func.isRequired,
};

export default AddMovieModal;
