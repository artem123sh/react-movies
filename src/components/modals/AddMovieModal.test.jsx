import React from 'react';
import { render } from '@testing-library/react';
import AddMovieModal from './AddMovieModal';

describe('AddMovieModal', () => {
  test('renders AddMovieModal\'s snapshot', () => {
    const { asFragment } = render(<AddMovieModal onClose={() => {}} handleMovieAdd={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
