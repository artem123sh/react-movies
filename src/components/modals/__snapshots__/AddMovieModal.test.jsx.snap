// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieModal renders AddMovieModal's snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ kYxXGN"
  />
  <div
    class="sc-AxiKw ghPDlE"
  >
    <div>
      <div
        class="sc-AxhUy fSLCtT"
      >
        <span
          class="sc-AxhCb fGYvSv"
        >
          â•³
        </span>
      </div>
      <div
        class="sc-AxheI iMsYLJ"
      >
        <span
          class="sc-AxjAm sc-AxgMl YYvbv"
        >
          add movie
        </span>
        <div
          class="sc-Axmtr eqaOhC"
        >
          <form
            action="#"
          >
            <label
              class="sc-fzqBZW sc-fzqNJr fRGmBX"
              for="title"
            >
              Title
            </label>
            <input
              class="sc-fzozJi sc-fzoLag bfKfQ"
              id="title"
              placeholder="Title"
              type="text"
              value=""
            />
            <label
              class="sc-fzqBZW sc-fzqNJr fRGmBX"
              for="release_date"
            >
              Release Date
            </label>
            <div
              class="sc-fzoXzr ejcWPC"
            >
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <input
                    class="sc-fzoLsD eupnYL sc-fzoyAV fnVNcw"
                    placeholder="Select Date"
                    value=""
                  />
                </div>
              </div>
            </div>
            <label
              class="sc-fzqBZW sc-fzqNJr fRGmBX"
              for="poster_path"
            >
              Movie Url
            </label>
            <input
              class="sc-fzozJi sc-fzoLag bfKfQ"
              id="poster_path"
              placeholder="Movie Url here"
              type="text"
              value=""
            />
            <label
              class="sc-fzqBZW sc-fzqNJr fRGmBX"
              for="genres"
            >
              Genre
            </label>
            <div
              class="sc-fzplWN dcwDRX"
            >
              <div>
                <div
                  class="sc-fznKkj crAqQg"
                >
                  <input
                    autocomplete="off"
                    class="sc-fzozJi sc-fznZeY ckxiKz"
                    id="genres"
                    placeholder="Select Genre"
                    readonly=""
                    type="text"
                    value=""
                  />
                  <svg
                    class="sc-fzokOt gafkUS"
                    fill="#F65261"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <label
              class="sc-fzqBZW sc-fzqNJr fRGmBX"
              for="overview"
            >
              Overview
            </label>
            <input
              class="sc-fzozJi sc-fzoLag bfKfQ"
              id="overview"
              placeholder="Overview here"
              type="text"
              value=""
            />
            <label
              class="sc-fzqBZW sc-fzqNJr fRGmBX"
              for="runtime"
            >
              Runtime
            </label>
            <input
              class="sc-fzozJi sc-fzoLag bfKfQ"
              id="runtime"
              placeholder="Runtime here"
              type="number"
              value=""
            />
            <label
              class="sc-fzqBZW sc-fzqNJr fRGmBX"
              for="tagline"
            >
              Tagline
            </label>
            <input
              class="sc-fzozJi sc-fzoLag bfKfQ"
              id="tagline"
              placeholder="Tagline here"
              type="text"
              value=""
            />
            <div
              class="sc-fznxsB cGbnwj"
            >
              <button
                class="sc-AxmLO sc-fznJRM eTUyfJ"
                type="reset"
              >
                Reset
              </button>
              <button
                class="sc-AxmLO sc-fznJRM bLGorp"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
